video_loader.py

```diff
@@ -138,6 +145,25 @@ def extract(self, docs: Optional[DocumentArray] = None, parameters: Dict = {}, *
                     chunk.blob, chunk.tags['sample_rate'] = audio, sr
                     doc.chunks.append(chunk)
 
+                # add subtitle ad chunks to the Document, modality='text'
+                if 'text' in self._modality:
+                    ffmpeg_subtitle_args = deepcopy(self._ffmpeg_subtitle_args)
+                    ffmpeg_subtitle_args.update(
+                        parameters.get('ffmpeg_subtitle_args', {})
+                    )
+                    subtitles = self._convert_video_uri_to_subtitle(
+                        source_fn, doc.uri, ffmpeg_subtitle_args, tmpdir
+                    )
+                    # subtitles = self._load_subtitles(doc.uri)
+                    for idx, (beg, end, s) in enumerate(subtitles):
+                        len_tokens = len([t for t in s.split(' ')])
+                        chunk = Document(id=f'{doc.id}_{idx}', text=s, modality='text')
+                        chunk.tags['beg_in_seconds'] = beg
+                        chunk.tags['end_in_seconds'] = end
+                        chunk.location.append(int(beg * 1000))  # location in milliseconds
+                        chunk.location.append(int(end * 1000))  # location in milliseconds
```

Comment:
This information is redundant with `tags['beg_in_seconds']`
```suggestion
                        chunk.location.append(idx)  # index of the subtitle in the video
```